src/server/server.js

```diff
@@ -151,19 +151,24 @@ function isAuthenticated(req, res, next) {
 
 app.use("/public", express.static(path.join(__dirname, "public")));
 
+let redirectedUrl = "/"
+
 app.get(
-  "/auth/provider",
-  passport.authenticate("provider", {
-    scope: "openid",
-  })
+  "/auth/provider", function (req, res, next){
+    redirectedUrl = req.headers.referer
```

Comment:
Hola! Instead of directly passing `req.headers.referer`, I'd suggest to parse out the pathname of the referer, i.e. you do `const referer = new URL(req.headers.referer)` and then do `redirectedUrl = referer.pathname`. This at least makes it impossible for evil eve to redirect to a different hostname.